distributed/tests/test_scheduler.py

```diff
@@ -925,47 +926,6 @@ async def test_delete(c, s, a):
     s.report_on_key(key=x.key)
 
 
-@gen_cluster()
-async def test_filtered_communication(s, a, b):
-    c = await connect(s.address)
-    f = await connect(s.address)
-    await c.write({"op": "register-client", "client": "c", "versions": {}})
-    await f.write({"op": "register-client", "client": "f", "versions": {}})
-    await c.read()
-    await f.read()
-
-    assert set(s.client_comms) == {"c", "f"}
-
-    await c.write(
-        {
-            "op": "update-graph",
-            "tasks": {"x": dumps_task((inc, 1)), "y": dumps_task((inc, "x"))},
-            "dependencies": {"x": [], "y": ["x"]},
-            "client": "c",
-            "keys": ["y"],
-        }
-    )
-
-    await f.write(
-        {
-            "op": "update-graph",
-            "tasks": {
-                "x": dumps_task((inc, 1)),
-                "z": dumps_task((operator.add, "x", 10)),
-            },
-            "dependencies": {"x": [], "z": ["x"]},
-            "client": "f",
-            "keys": ["z"],
-        }
-    )
```

Comment:
This is why I dislike white-box tests.  There was a lot of this in early days.

Reply:
FWIW I encountered the same test over https://github.com/dask/distributed/pull/7502#discussion_r1087713129 already
I believe the condition this is constructing is actually not possible when using the Client API

> This is why I dislike white-box tests

I actually don't consider this a "white box" test. We're not asserting on internal state of the object but are rather using a very low level API where instead a high level API would've been more useful. A different way to look at this is that we actually don't care about the low level API that much.